### PR TITLE
Fix reference implementation for Unique

### PR DESCRIPTION
**Pull request by neNasko1**
### Description
This PR uses the [recommended way](https://numpy.org/doc/stable/reference/generated/numpy.unique.html) to ensure compatibility with `numpy.unique`.

Additionally the `strict` parameter is turned on for `numpy.testing.assert_allclose` to ensure that the `dtype`-s and `shape`-s match, this fortunately does not introduce additional test failures. 

### Motivation and Context
The [shape inference](https://github.com/onnx/onnx/blob/b50fb874f71325030d93dae4ae04a2647de7a30e/onnx/defs/tensor/defs.cc#L3371) defined for `Unique`, tells us the fact that the `inverse_indices` output should be 1d. This was not the case when the input was of length 1, since the `squeeze` made it a scalar. 

This mismatch between the reference implementation and `onnxruntime` was found by running the [array-api](https://data-apis.org/array-api/latest/) test suite through [ndonnx](https://github.com/Quantco/ndonnx) on both.

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]
Thank you!

File: onnx/backend/test/runner/__init__.py [hunk 1]
```diff
@@ -215,13 +215,15 @@ def assert_similar_outputs(
                     if ref_outputs[i].tolist() != outputs[i].tolist():
                         raise AssertionError(f"{ref_outputs[i]} != {outputs[i]}")
                     continue
+                np.testing.assert_equal(outputs[i].dtype, ref_outputs[i].dtype)
+                np.testing.assert_array_equal(outputs[i].shape, ref_outputs[i].shape)
```
Comment (justinchuby) [inline onnx/backend/test/runner/__init__.py:219]: Probably good to provide a message telling users this is testing for the shape

Comment (neNasko1) [reply onnx/backend/test/runner/__init__.py]: Thanks for the suggestion, just pushed a fix. Here is how it looks like in a case of failure after the last commit:
```console
E   AssertionError:
E   Arrays are not equal
E   Output 0 has incorrect shape
E   (shapes (0,), (3,) mismatch)
E    x: array([], dtype=float64)
E    y: array([3, 4, 5])
```

**Review by justinchuby — APPROVED** [review]

**Comment by neNasko1** [comment]
Thanks for the review!

**Pull request merged**